src/main/java/com/parkhomovsky/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -112,7 +112,18 @@ public ResponseEntity<Object> handleEntityNotFoundException(
         String errorMessage = "An error occurred while proceeding data from database. "
                 + "Nothing found by provided parameters";
         ErrorResponseDto errorResponse =
-                getErrorMessageBody(errorMessage, rootCause, HttpStatus.BAD_REQUEST);
+                getErrorMessageBody(errorMessage, rootCause, HttpStatus.NOT_FOUND);
+        return new ResponseEntity<>(errorResponse, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(EmptyShoppingCartException.class)
+    public ResponseEntity<Object> handleEmptyShoppingCartException(
+            EmptyShoppingCartException ex
+    ) {
+        Throwable rootCause = ExceptionUtils.getRootCause(ex);
+        String errorMessage = "An error occurred while proceeding shopping cart data.";
```

Comment:
You don't need to create a variable each time error occurs, make it inline or extract to constant.

Reply:
not resolved